src/diffpy/utils/scattering_objects/diffraction_objects.py

```diff
@@ -362,18 +362,125 @@ def tth_to_q(self):
         pre_factor = (4 * np.pi) / wavelength
         return pre_factor * np.sin(two_theta / 2)
 
+    def q_to_d(self):
+        r"""
+        Helper function to convert q to d using :math:`d = \frac{2 \pi}{q}`
+
+        adds a small value (epsilon = 1e-10) to `q` where `q` is close to zero to avoid division by zero
+
+        Parameters
+        ----------
+        q : array
+            An array of :math:`q` values
+
+        Returns
+        -------
+        d : array
+            An array of :math:`d` values in the inverse of the units of ``wavelength``
+        """
+        epsilon = 1e-10
+        q = np.asarray(self.on_q[0])
+        q = np.where(np.abs(q) < epsilon, q + epsilon, q)
+        return (2 * np.pi) / q
+
+    def d_to_q(self):
+        r"""
+        Helper function to convert d to q using :math:`q = \frac{2 \pi}{d}`
+
+        adds a small value (epsilon = 1e-10) to `d` where `d` is close to zero to avoid division by zero
+
+        Parameters
+        ----------
+        d : array
+            An array of :math:`d` values
+
+        Returns
+        -------
+        q : array
+            An array of :math:`q` values in the inverse of the units of ``wavelength``
+        """
+        epsilon = 1e-10
+        d = np.asarray(self.on_d[0])
+        d = np.where(np.abs(d) < epsilon, d + epsilon, d)
```

Comment:
Added d to a small number to avoid dividing by 0, which would cause q to become some arbitrary large number (as commented later in the test function). Not sure if this is a good way to address this problem - do we want to just raise error in these cases?